modules/admin-server/src/main/kotlin/com/sphereon/oid/fed/server/admin/handlers/logger/FileLoggerHandler.kt

```diff
@@ -2,31 +2,72 @@ package com.sphereon.oid.fed.server.admin.handlers.logger
+                    try {
+                        Files.createDirectories(parent.toPath())
+                        println("Successfully created log directory")
+                    } catch (e: Exception) {
+                        val msg = "Failed to create log directory: ${parent.absolutePath}. Error: ${e.message}"
+                        println(msg)
+                        throw IllegalStateException(msg, e)
+                    }
+                }
+
+                // Check directory permissions
+                if (!parent.canWrite()) {
+                    val msg = "No write permission for log directory: ${parent.absolutePath}"
+                    println(msg)
+                    throw IllegalStateException(msg)
                 }
             }
-            if (!logFile.exists() && !logFile.createNewFile()) {
-                throw IllegalStateException("Failed to create log file: ${logFile.absolutePath}")
+
+            if (!logFile.exists()) {
+                println("Log file doesn't exist, attempting to create: ${logFile.absolutePath}")
+                try {
+                    logFile.createNewFile()
+                    println("Successfully created log file")
+                } catch (e: Exception) {
+                    val msg = "Failed to create log file: ${logFile.absolutePath}. Error: ${e.message}"
+                    println(msg)
+                    throw IllegalStateException(msg, e)
+                }
             }
+
             if (!logFile.canWrite()) {
-                throw IllegalStateException("Log file is not writable: ${logFile.absolutePath}")
+                val msg = "Log file is not writable: ${logFile.absolutePath}"
+                println(msg)
+                throw IllegalStateException(msg)
             }
+
+            println("Successfully initialized log file handler")
         } catch (e: SecurityException) {
-            throw IllegalStateException("Security violation while setting up log file: ${logFile.absolutePath}", e)
+            val msg = "Security violation while setting up log file: ${logFile.absolutePath}"
+            println("$msg. Error: ${e.message}")
+            throw IllegalStateException(msg, e)
         } catch (e: Exception) {
-            throw IllegalStateException("Failed to initialize log file: ${logFile.absolutePath}", e)
+            val msg = "Failed to initialize log file: ${logFile.absolutePath}"
+            println("$msg. Error: ${e.message}")
+            throw IllegalStateException(msg, e)
         }
     }
 
     override fun log(event: Logger.LogEvent) {
         synchronized(this) {
-            logFile.appendText("${event.formattedMessage}\n")
+            try {
+                logFile.appendText("${event.formattedMessage}\n")
+            } catch (e: Exception) {
+                println("Failed to write to log file: ${logFile.absolutePath}. Error: ${e.message}")
+                // Consider implementing a fallback logging mechanism here
```

Comment:
Should we have a ticket here for it
```suggestion
                // Implementation of a fallback error or log tracer: TICKET: XXX-123
```

Reply:
This is a simple implementation of the Logger.LogWritter interface. Not a priority.